TLM/TLM/UI/SubTools/PrioritySigns/Overlay.cs

```diff
@@ -0,0 +1,516 @@
+namespace TrafficManager.UI.SubTools.PrioritySigns {
+    using ColossalFramework;
+    using TrafficManager.API.Manager;
+    using TrafficManager.Manager.Impl;
+    using TrafficManager.State;
+    using TrafficManager.UI.Textures;
+    using TrafficManager.Util;
+    using UnityEngine;
+
+    /// <summary>
+    /// Class handles rendering of priority signs overlay.
+    /// Create one and set its fields before calling DrawSignHandles
+    /// </summary>
+    public struct Overlay {
+        private const float JUNCTION_RESTRICTIONS_SIGN_SIZE = 80f;
+
+        private readonly TrafficManagerTool mainTool_;
+        private readonly bool debug_;
+        private readonly bool handleClick_;
+
+        public bool ViewOnly;
+
+        /// <summary>
+        /// Handles layout for the Junction Restriction signs being rendered.
+        /// One <see cref="SignsLayout"/> is created per junction.
+        /// Defines basis of rotated coordinate system, aligned somewhere near the node center,
+        /// and directed along the road segment.
+        /// </summary>
+        private struct SignsLayout {
+            private Vector3 zero_;
+
+            /// <summary>Unit vector perpendicular to the vector towards the segment center.</summary>
+            public Vector3 yBasis;
+
+            /// <summary>Unit vector towards the segment center.</summary>
+            public Vector3 xBasis;
+
+            private readonly int signsPerRow_;
+            private readonly bool viewOnly_;
+
+            /// <summary>Zoom level inherited from the MainTool.</summary>
+            private readonly float baseZoom_;
+
+            private float signSize_;
+
+            /// <summary>Horizontal position when placing signs in a grid.</summary>
+            private int xPosition;
+
+            /// <summary>Vertical position when placing signs in a grid.</summary>
+            private int yPosition;
+
+            public SignsLayout(Vector3 nodePos,
+                               Vector3 yBasis,
+                               int signsPerRow,
+                               bool viewOnly,
+                               float baseZoom) {
+                this.signsPerRow_ = signsPerRow;
+                this.viewOnly_ = viewOnly;
+                baseZoom_ = baseZoom;
+
```

Comment:
`xBasis` becomes `yBasis` and `yBasis` becomes vector perpendicular to `yBasis`. Don't you think it would be better to use more consistent naming.
also this code is faster and much easier to read than cross product.(just a suggestion)
```C#
this.yBasis = new Vector3(-xBasis.z, 0, xBasis.x);// rotate 90 degree
```

Reply:
you don't need the z dimension so y not to use 2D vectors? use `VectorUtils.XZ()` 
be aware of auto-casting `Vector2 v = vector3D` which is equivalent to `VectorUtils.XY()` and not what we want.